hnswlib/hnswalg.h

```diff
@@ -799,10 +842,64 @@ class HierarchicalNSW : public AlgorithmInterface<dist_t> {
     }
 
 
-    /**
-        * Adds point. Updates the point if it is already in the index
+    /*
+    * Adds point and replaces previously deleted point if any, updating it with new point
+    * If deleted point was replaced returns its label, else returns label of added or updated point
+    * 
+    * Note:
+    *  Methods that can work with deleted elements unmarkDelete and addPoint are not safe to use
+    *  with this method, because addPointToVacantPlace removes deleted elements from the index. 
+    */
+    labeltype addPointToVacantPlace(const void* data_point, labeltype label) {
```

Comment:
Can we make it a flag to `addPoint` instead of a separate function? 
I guess users who opt to use the replacement might still want to use fast insertions in some cases.  
A flag might cause to break brute force, but I think we can throw an exception that it is unsupported.